### ISSUE TITLE
AKS discovery issue

### ISSUE THREAD

**Issue by satishblr** [newrelic/nri-discovery-kubernetes#12]
My config looks like this

```kafka-config.yml
---
# Run auto discovery to find pods with label "app=kafka"
# https://docs.newrelic.com/docs/integrations/host-integrations/installation/container-auto-discovery
discovery:
  command:
    # Run NRI Discovery for Kubernetes
    # https://github.com/newrelic/nri-discovery-kubernetes
    exec: /var/db/newrelic-infra/nri-discovery-kubernetes --tls --port 10250
    match:
      label.app: cp-kafka
integrations:
  # This instance gives an example of autodiscovery of brokers with a bootstrap broker
  - name: nri-kafka
    env:
      # A cluster name is required to uniquely identify this collection result in Insights
      CLUSTER_NAME: cr-shared-npd-kafka

      AUTODISCOVER_STRATEGY: bootstrap

      # Bootstrap broker arguments. These configure a connection to a single broker. The rest of the brokers in the cluster
      # will be discovered using that connection.
      BOOTSTRAP_BROKER_HOST: ${discovery.ip}
      # Ports can be autodiscovered. You can use an ordinal number like ${discovery.ports.0}
      # for the first discovered port (they are sorted in ascending order)
      # E.g. if the pod ports are {9092, 9999}, then ${discovery.ports.0} has the value 9092 and
      # ${discovery.ports.1} equals 9999
      # Named ports can also be used like this ${discovery.ports.<name>}. E.g. ${discovery.ports.kafka}
      BOOTSTRAP_BROKER_KAFKA_PORT: 9093
      BOOTSTRAP_BROKER_KAFKA_PROTOCOL: SSL # Currently support PLAINTEXT and SSL

      BOOTSTRAP_BROKER_JMX_PORT: 5555
      # JMX user and password default to `default_jmx_user` and `default_jmx_password` if unset
      #BOOTSTRAP_BROKER_JMX_USER: username
      #BOOTSTRAP_BROKER_JMX_PASSWORD: password
      TLS_CA_FILE: /secret/kafka-ca-file
      TLS_CERT_FILE: /secret/kafka-cert-file
      TLS_KEY_FILE: /secret/kafka-key-file
      # Only collect metrics from the bootstrap broker configured. The integration will not attempt to collect metrics
      # for any other broker, nor will it collect cluster-level metrics like topic metrics. This is useful for things
      # like deployment to kubernetes, where a single integration instance is desired per broker.
      # In order to collect metrics from a single broker, you may need to set the KAFKA_ADVERTISED_HOST_NAME in your
      # deployment file to the same value as in the BOOTSTRAP_BROKER_HOST config option. E.g. using the podIP:
      # - name: KAFKA_ADVERTISED_HOST_NAME
      #   valueFrom:
      #     fieldRef:
      #       apiVersion: "v1"
      #       fieldPath: "status.podIP"
      LOCAL_ONLY_COLLECTION: true

      # See above for more information on topic collection
      COLLECT_BROKER_TOPIC_DATA: true
      TOPIC_MODE: all
      COLLECT_TOPIC_SIZE: true

      METRICS: 1```

when the command runs getting this error

`2020/09/02 21:50:26 failed to connect to Kubernetes: failed to execute request against kubelet: Get https://aks-default-29900020-vmss000000:10250/pods: dial tcp: lookup aks-default-29900020-vmss000000 on 192.168.204.10:53: server misbehaving`

am I missing anything in config? 

Thanks
Satish

**Comment by josemore** [comment]
Thanks for reporting the issue @satishblr 
There's a known-issue in our backlog where the node name can't be resolved via DNS though customers reported that localhost works (this seems to be the same issue). 
A fix is expected in a future release but we can't commit to a date at this moment.

**Comment by tobybro** [comment]
Any updates here? Is switching to localhost a workaround and if so how do I do that?

**Comment by josemore** [comment]
We'll discuss and work on an fix for this problem on #17
